controller/run.go

```diff
@@ -56,33 +56,30 @@ func RunUnmanaged(cfg config.ShipperRootConfig) error {
 	runner := publisherserver.NewOutputServer()
 	err = runner.Start(cfg)
 	if err != nil {
-		return err
+		return fmt.Errorf("error starting publisher server: %w", err)
 	}
 
 	srv := grpcserver.NewGRPCServer(runner)
-
-	done := make(doneChan)
-	go func() {
-		err = srv.Start(creds, cfg.Shipper.Server.Server)
-		done <- struct{}{}
-	}()
+	err = srv.Start(creds, cfg.Shipper.Server.Server)
 
 	// On termination signals, gracefully stop the shipper
 	sigc := make(chan os.Signal, 1)
 	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
 
-	sig := <-sigc
-	switch sig {
-	case syscall.SIGINT, syscall.SIGTERM:
-		log.Debug("Received sigterm/sigint, stopping")
-	case syscall.SIGHUP:
-		log.Debug("Received sighup, stopping")
+	select {
+	case sig := <-sigc:
+		switch sig {
+		case syscall.SIGINT, syscall.SIGTERM:
+			log.Debug("Received sigterm/sigint, stopping")
```

Comment:
```suggestion
			log.Info("Received sigterm/sigint, stopping")
```

I think the reason for stopping should be at the Info level so it is always obvious. It won't happen frequently enough to count as log spam.